ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,57 @@ def decorated(*args, **kwargs):
     return decorated
 
 
-# TODO resource route for development purpose - REMOVE LATER
-# Protected resource
-@app.route("/resource", methods=["POST"])
+# ******************** Credential Store API ******************** #
+
+
+# Credential Store API - GET Method - Get list of all credentials
+@app.route("/credential_store", methods=["GET"])
+@token_required
+def credential_store_endpoint(current_user):
+    """
+    Return a list of credentials and their information in JSON format.
+
+    :param current_user: Information of the current_user based on the request's JWT token
+    """
+
+    from GangaCore.GPI import credential_store
+
+    # Store credential store info in a list
+    credential_info_list = []
+    try:
+        for c in credential_store:
+            credential_info = {}
+            credential_info["location"] = str(c.location)
+            credential_info["time_left"] = str(c.time_left())
+            credential_info["expiry_time"] = str(c.expiry_time())
+            credential_info["is_valid"] = str(c.is_valid())
+            credential_info["exists"] = str(c.exists())
+            credential_info_list.append(credential_info)
```

Comment:
Could this be built in some dynamic way (to avoid the hard coding of the method names and attributes)?

Reply:
I can shorten this, but this is the monitoring information that is fixed and would be need for the front end. I could have made another route more generic use, in a sense users could just ask what they want like in case of `job` API but I looked into `ICredentialInfo` it didn't have a `schema` or `exportmethods`.